src/Mod/Part/App/FCBRepAlgoAPI_BooleanOperation.cpp

```diff
@@ -109,18 +109,18 @@ void FCBRepAlgoAPI_BooleanOperation::Build() {
         RecursiveAddArguments(myOriginalTools.First());
         if (!myTools.IsEmpty()) {
             myOperation = BOPAlgo_FUSE; // fuse tools together
-            Build();
+            BRepAlgoAPI_BooleanOperation::Build();
             myOperation = BOPAlgo_CUT; // restore
             myArguments = myOriginalArguments;
             if (IsDone()) {
                 myTools.Append(myShape);
-                Build(); // cut with fused tools
+                BRepAlgoAPI_BooleanOperation::Build(); // cut with fused tools
```

Comment:
I think the intension was to do a recursion here as well. It basically tries again after fusing the tools.
```suggestion
                Build(); // cut with fused tools
```